src/utilities/utils.js

```diff
@@ -0,0 +1,53 @@
+const sortPrice = (productList, sortBy) => {
+    console.log("from context",productList,sortBy)
+        let updatedList = [...productList];
+       
+        if (sortBy === "HIGH_TO_LOW") {
+            console.log("updatedList", updatedList.sort((a, b) => b.price- a.price))
+          return updatedList.sort((a, b) => b.price - a.price);
+        }
+        if (sortBy === "LOW_TO_HIGH") {
+          return updatedList.sort((a, b) => a.price - b.price);
+        }
+       
+        return updatedList;
+ };
+
+const filteredProduct = (productList, state)=>{
+    return productList.filter(item => state.showAllProduct ? true : !item.isSoldOut)
+}
+
+const filterPrice = (productList, priceValue)=>{
+let updatedList = [...productList]
+if(priceValue){
+    return updatedList.filter(item => Number(item.price) <= Number(priceValue))
+}
+return updatedList
+    
+}
+
+const filterCategory =(productList,state)=>{
+    const categoryList = state.sortByCategory;
+    return categoryList.length===0 ? productList:
+    productList.filter((item)=>categoryList.some((category)=>category===item.categoryName))
+}
+
+
+const filterRating = (productList, rating) => {
+    if(rating === "FOUR_STARS_AND_ABOVE"){
+        return productList.filter(item => item.rating >= 4)
+    }
+    if(rating === "THREE_STARS_AND_ABOVE"){
+        return productList.filter(item=> item.rating >= 3 && item.rating <= 4)
```

Comment:
There seems to be a logical error in your code, as selecting filter 3 stars and above should show products having rating above 3 stars as well as above 4 stars. Similarly for others as well. However, that is not the case as you can see in the screenshot below
![Screenshot (137)](https://user-images.githubusercontent.com/30971358/160785665-c86068ca-ea0e-4b63-a539-5564c004a165.png)

You can fix this by removing the condition after && from your rating filter statement in this one as well as in the filters below this.